party_time/party_time.lua

```diff
@@ -0,0 +1,346 @@
+local os = require('os')
+local ui = require('ui')
+local sets = require("sets")
+local string = require('string')
+local command = require('command')
+local packets = require('packets')
+local settings = require('settings')
+local treasure = require('treasure')
+
+local defaults = {
+    ui = {
+        x = 145,
+        y = 440,
+        enabled = true,
+    },
+    auto = {
+        accept_invites = true,   -- also for auto sending invites upon party requests
+        decline_invites = false, -- also ignores party requests from players on blacklists
+    },
+    blacklist = {}, 
+    whitelist = {},
+}
+settings.settings_change:register(function(options)
+    options.blacklist = sets(options.blacklist)
+    options.whitelist = sets(options.whitelist)
+end)
+options = settings.load(defaults)
+options.blacklist = sets(options.blacklist)
+options.whitelist = sets(options.whitelist)
```

Comment:
Don't need these anymore if you have them defined prior to the `settings.load` call, as that will invoke the registered function itself.

Reply:
without these i get error for adding a set to a non-set, this is what i changed last night to get it working.

Reply:
Right. You need to define the defaults with `blacklist = sets({})` instead of `= {}`, as that makes them sets, and then you can use all set operations on them.